### PR TITLE
Add event priorities, refactor transfer worker, some cleanup

### PR DESCRIPTION
**Pull request by wchill**
### Short Description:

The event manager now supports assigning priorities to events. Negative priorities run first, while positive priorities run last. This allows for pipeline-based processing of event arguments, which leads to the next change. Any event handler that does not specify a priority automatically gets assigned a priority of 0.

`InitialTransferWorker` has been refactored into a plugin. This plugin is a good demonstration of the improved event system, where a filter function `filter_pokemon` registers with priority -1000 and runs before the actual transferring code `transfer_pokemon`, which is registered with priority 1000. Here, `filter_pokemon` will take a list of Pokemon to transfer (calling the API if no list was provided) and filter the list according to the user's configuration settings, returning the new list. The event manager will then update its dictionary of named arguments that the event was triggered with and call the next handler, `transfer_pokemon`. If nothing is returned from a function, then the dictionary remains unchanged.

Some bits and pieces of code were also cleaned up.
### Changes:
- Event manager supports priorities/pipeline-based processing
- `InitialTransferWorker` refactored into plugin
- General cleanup

@OpenPoGo/maintainers

### CODE REVIEW

**Comment by jtdroste** [comment]
I'm curious - why are we going with greatest number first?  I say remove the `* -1` you do in the fire function.

This would also require changing the priority on the `transfer_pokemon` plugin

**Comment by wchill** [comment]
Priority shouldn't really be a negative number, and so doing it this way ensures that a handler can always have something running before it in the pipeline no matter what its priority.

**Comment by jtdroste** [comment]
Talked about on slack.

👍 from me now.  Negative number priorities run first, positive numbers run last.  0 is the default.

[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/126/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

**Comment by MaxLeiter** [comment]
:+1: from me, would be useful to maybe have a wiki page documenting this? (basically the initial PR description)

@wchill, will leave for you to merge when you feel it's ready.
[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/126/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

File: plugins/transfer_pokemon/__init__.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        # ignore deployed pokemon
+        if pokemon.deployed_fort_id is not None:
+            continue
+
+        pokemon_num = pokemon.pokemon_id
+        if pokemon_num not in indexed_pokemon:
+            indexed_pokemon[pokemon_num] = list()
+        indexed_pokemon[pokemon_num].append(pokemon)
+
+    if bot.config.cp and False:
```
Comment (andythorne) [inline plugins/transfer_pokemon/__init__.py:31]: `and False` - Did you mean to leave this in?

Comment (wchill) [inline plugins/transfer_pokemon/__init__.py:31]: Oops, I did not mean to leave that in. File a PR for that please @andythorne and I'll approve it; I'm on mobile right now so I can't do it myself.

Comment (andythorne) [inline plugins/transfer_pokemon/__init__.py:31]: No worries, i'm just updating https://github.com/OpenPoGo/OpenPoGoBot/pull/115 atm, so i'll roll it into there :)

File: plugins/transfer_pokemon/__init__.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        ignore_list = bot.config.ign_init_trans.split(',')
```
Comment (1karu5) [inline plugins/transfer_pokemon/__init__.py:32]: example config (config.json.example) for the ign_init_trans part?

**Pull request merged**